pkg/util/ecs/common.go

```diff
@@ -110,18 +117,34 @@ func convertMetaV2Container(c v2.Container) *containers.Container {
 		container.StartedAt = startedAt.Unix()
 	}
 
-	if l, found := c.Limits["cpu"]; found && l > 0 {
+	if l, found := c.Limits[cpuKey]; found && l > 0 {
 		container.Limits.CPULimit = float64(l)
+	} else if l, found := taskLimits[cpuKey]; found && l > 0 {
+		container.Limits.CPULimit = formatTaskCPULimit(l)
 	} else {
 		container.Limits.CPULimit = 100
 	}
-	if l, found := c.Limits["memory"]; found && l > 0 {
-		container.Limits.MemLimit = l
+
+	if l, found := c.Limits[memoryKey]; found && l > 0 {
+		container.Limits.MemLimit = formatMemoryLimit(l)
+	} else if l, found := taskLimits[memoryKey]; found && l > 0 {
+		container.Limits.MemLimit = formatMemoryLimit(uint64(l))
 	}
 
 	return container
 }
 
+func formatTaskCPULimit(val float64) float64 {
+	// The ECS API exposes the task CPU limit with the format: 0.25, 0.5, 1, 2, 4
+	// We convert the value to CPU units
+	return val * 1024 / 1000 * 100
```

Comment:
As process expects the value in Hz (1 vCPU = 100), for a container limits set to 512 or ECS task limit set to 0.5 we should return 50, which means:
-> taskLimit * 100
-> containerLimit / 1024 * 100

If we want to expose our metric in shares (so reporting 512) we can keep the value separately, but we should keep process in `Hz`

Reply:
@shang-wang any thoughts on this?

Reply:
Just chatted with Vincent as I'm not super familiar with the concept but yea, that makes sense. We should unify the unit as much as we can, thanks.